packages/plugin-core/src/WorkspaceWatcher.ts

```diff
@@ -100,4 +106,33 @@ export class WorkspaceWatcher {
     }
     return { changes };
   }
+
+  onDidOpenTextDocument(document: TextDocument) {
+    this._openedDocuments.set(document.uri.fsPath, document);
+  }
+
+  /** The documents that have been opened during this session that have not been viewed yet in the editor.
+   *
+   * Had to add undefined because
+   */
+  private _openedDocuments: Map<string, TextDocument> = new Map();
+
+  /** Do not use this function, please go to `WindowWatcher.onFirstOpen() instead.`
```

Comment:
Actually do we need the `workspace.onDidOpenTextDocument` event? Can we just move this openedDocument tracking into windowWatcher and only rely on the `window.onDidChangeActiveTextEditor` event?

Reply:
Unfortunately not. `window.onDidChangeActiveTextEditor` activates when switching between editors, so it would fire whenever the user switched to an already open note.

Worse, when `window.onDidChangeActiveTextEditor` activates `window.selection` is not set to the actual user selection in that window yet, so we can't even check if the user had selected something and switched out. 

And when `workspace.onDidOpenTextDocument` activates, the active editor has not switched yet, so we can't just grab the active editor at that point either.